src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -278,38 +277,30 @@ protected override TDeclarationNode AddNamespace<TDeclarationNode>(TDeclarationN
             CodeGenerationOptions options,
             CancellationToken cancellationToken)
         {
-            var memberDeclaration = destinationMember as MemberDeclarationSyntax;
-            if (memberDeclaration == null)
-            {
-                return destinationMember;
-            }
+            var s_generator = CSharpSyntaxGenerator.Instance;
+            var currentParameterList = s_generator.GetParameters(destinationMember);
 
-            var currentParameterList = memberDeclaration.GetParameterList();
             if (currentParameterList == null)
```

Comment:
`GetParameters` never returns null. 
[http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/CodeGeneration/CSharpSyntaxGenerator.cs,2593](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/CodeGeneration/CSharpSyntaxGenerator.cs,2593)

Reply:
I think `GetParameterList` is probably what you wanted

---
In reply to: [312287954](https://github.com/dotnet/roslyn/pull/37629#discussion_r312287954) [](ancestors = 312287954)